src/basic/pidfd-util.c

```diff
@@ -18,15 +19,12 @@
 
 static int have_pidfs = -1;
 
-static int pidfd_check_pidfs(void) {
+static int pidfd_check_pidfs(int fd) {
 
         if (have_pidfs >= 0)
                 return have_pidfs;
 
-        _cleanup_close_ int fd = pidfd_open(getpid_cached(), 0);
-        if (fd < 0)
-                return -errno;
-
+        /* Allocate a pidfd to ourselves if no fd passed */
```

Comment:
Hmm? Don't grok this comment? There's no logic corresponding to this?

Reply:
oops, before removing the allocation entirely i made it optional, forgot to remove the comment

Reply:
BTW, I implemented things this way in #35242, but @yuwata was against it: https://github.com/systemd/systemd/pull/35242#discussion_r1863493996

Reply:
Yeah, I do not like this. At least, strong warning comment should be added.

Reply:
ok, i added such a comment now, and i renamed the parameter to indicate clearly that it's supposed to be a pidfd